### PR TITLE
feat: initial implementation

### PR DESCRIPTION
**Pull request by erik-slovak**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  "scripts": {
+    "lint": "eslint --format=node_modules/eslint-formatter-pretty --ext .jsx,.js,.ts,.tsx .",
+    "fixcode": "eslint --format=node_modules/eslint-formatter-pretty --ext .jsx,.js,.ts,.tsx . --fix",
+    "git-hook": "yarn -s lint",
+    "ncu-all": "ncu -i",
+    "compile": "tsc",
+    "prepare": "husky install && yarn compile",
+    "sanity": "yarn lint && tsc --noEmit && yarn outdated && echo 'success'",
+    "semantic-release": "semantic-release",
+    "test": "yarn compile && cp -r lib node_modules/eslint-plugin-txo-typescript/ && yarn lint"
```
Comment (erik-slovak) [inline package.json:26]: @rostislav-simonik please check if this is okay, at the time of `yarn install` the `lib` folder doesn't exist yet, so the CI check fails when running `yarn test`

Comment (rostislav-simonik) [reply package.json:26]: yarn install --dev file:. instead

File: test/Sample.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * @Author: Erik Slovak <erik.slovak@technologystudio.sk>
+ * @Date: 2023-07-11T12:07:09+02:00
+ * @Copyright: Technology Studio
+**/
+
+import { Log } from './LogMock'
+
+// eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars, txo-typescript/correct-log-path
```
Comment (rostislav-simonik) [inline test/Sample.ts:9]: export them to avoid unnecessary eslint disables

File: package.json (hunk 1 above)
Comment (rostislav-simonik) [reply package.json]: ok, we use copy due broken yarn install for local package because  it copies also node_modules

**Review by rostislav-simonik — APPROVED** [review]

**Pull request merged**